pkg/internal/vault/vault.go

```diff
@@ -109,7 +110,7 @@ func (v *Vault) Sign(csrPEM []byte, duration time.Duration) (cert []byte, ca []b
 	request := v.client.NewRequest("POST", url)
 
 	if err := request.SetJSONBody(parameters); err != nil {
-		return nil, nil, fmt.Errorf("failed to build vault request: %s", err)
+		return nil, nil, cmErrors.NewInvalidData("failed to build vault request: %s", err)
```

Comment:
Not sure we should make these changes to CR controllers yet - they need some consideration (I'm not sure I agree with setting InvalidRequest=true for the cases here in this file for example)